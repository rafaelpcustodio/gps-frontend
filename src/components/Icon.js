import PropTypes from 'prop-types'
import React from 'react'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const Icon = ({ color, className, icon, size }) => (
    <FontAwesomeIcon 
        className={ className } color ={ color } data-cy="icon" icon={ icon } size={ size }
    />)

Icon.defaultProps = {
    className: '',
    color: '',
    icon: '',
    size: 'lg'
}

Icon.propTypes = {
    className: PropTypes.string,
    color: PropTypes.string,
    icon: PropTypes.string.isRequired,
    size: PropTypes.string
}

export default Icon
