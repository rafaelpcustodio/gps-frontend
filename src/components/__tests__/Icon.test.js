import configureMockStore from 'redux-mock-store'
import { Map, Record } from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import Icon from '../Icon'

describe('Tests for Icon component', () => {
    let store = {}
    let mockStore = {}

    it('Should render without crash', () => {
        mockStore = configureMockStore()
        store = mockStore(Record({
            gpsReducers: Map({
            })
        })())

        const tree = renderer.create(
            <Provider store={ store }>
                <Icon
                    className={''}
                    color={''}
                    icon={{'iconName': 'caret-down', 'prefix': 'fas', 'icon':[320,512]}}
                    size={''}
                />
            </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
