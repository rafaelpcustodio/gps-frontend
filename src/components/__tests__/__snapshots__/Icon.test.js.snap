// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Icon component Should render without crash 1`] = `null`;
